src/reactjs/Hydrate.tsx

```diff
@@ -3,6 +3,11 @@ import React from 'react'
 import { hydrate, HydrateOptions } from '../core'
 import { useQueryClient } from './QueryClientProvider'
 
+const IsHydratingContext = React.createContext(false)
+
+export const useIsHydrating = () => React.useContext(IsHydratingContext)
```

Comment:
Since we are at least _aiming_ at putting hydration data on a context now to avoid async hydration, we might end up not needing this for the regular hydration case. If that is the case, we might want to extract this out into persist instead and call it something not related to hydration to avoid confusion. A bit too early to say for sure though, since we might still need it if having hydration data on a context is too hard or complex for some reason. 🤷😄